libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,555 @@
+  };
+
+  const parsedXML = parser.parse(xml, options);
+  if (_.has(parsedXML, 'summaryreport')) {
+    throw new Error('Current mapper does not accept sumarry reports');
+  }
+  const arrayedControls = _.get(parsedXML, 'detailedreport.severity').map(
+    (control: {category: unknown; level: string}) => {
+      if (Array.isArray(control.category)) {
+        return {level: control.level, category: control.category};
+      } else if (!control.category) {
+        return {level: control.level};
+      } else {
+        return {level: control.level, category: [control.category]};
+      }
+    }
+  );
+  _.set(parsedXML, 'detailedreport.severity', arrayedControls);
+
+  // Sets cwe and flaw keys to be arrays
+  // Moves staticflaws up one level for control mapping
+  _.set(parsedXML, '', staticflawmover(parsedXML));
+  // Moves cves up one level for control mapping
+  const len = _.get(
+    parsedXML,
+    'detailedreport.software_composition_analysis.vulnerable_components.component.length'
+  );
+  const cveArr = [];
+  for (let i = 0; i < len; i++) {
+    if (
+      _.get(
+        parsedXML,
+        `detailedreport.software_composition_analysis.vulnerable_components.component[${i}].vulnerabilities`
+      ) !== ''
+    ) {
+      cveArr.push(
+        _.get(
+          parsedXML,
+          `detailedreport.software_composition_analysis.vulnerable_components.component[${i}]`
+        )
+      );
+    }
+  }
+  _.set(
+    parsedXML,
+    'detailedreport.software_composition_analysis.vulnerabilities',
+    cveArr
+  );
+
+  //format software_composition_analysis for hdf
+  _.set(parsedXML, '', cvesformatter(parsedXML));
+  _.set(parsedXML, '', setcomponents(parsedXML));
+
+  return parsedXML;
+}
+
+function controlMappingCwe(severity: number) {
+  return {
+    id: {path: 'categoryid'},
+    title: {path: 'categoryname'},
+    desc: {transformer: formatDesc},
+    descriptions: [
+      {
+        data: {transformer: formatRecommendations},
+        label: 'fix'
+      }
+    ],
+    impact: impactMapping(severity),
+    refs: [],
+    tags: {
+      cweid: {transformer: formatCweData},
+      cweDescription: {transformer: formatCweDesc},
+      nist: {transformer: nistTag}
+    },
+    source_location: {
+      ref: {
+        path: 'staticflaws.flaw',
+        transformer: formatSourceLocation
+      },
+    },
+    results: [
+      {
+        path: 'staticflaws.flaw',
+        status: ExecJSON.ControlResultStatus.Failed,
+        code_desc: {transformer: formatCodeDesc},
+        run_time: 0,
+        start_time: {path: '$.detailedreport.first_build_submitted_date'},
+        message: {path: 'exploitability_adjustments.exploitability_adjustment.note'}
+      }
+    ]
+  };
+}
+function controlMappingCve() {
+  return {
+    id: {path: 'cve_id'},
+    title: {path: 'cve_id'},
+    desc: {path: 'cve_summary'},
+    impact: {path: 'impact'},
+    refs: [],
+    tags: {
+      cwe: {path: 'cwe_id'},
+      nist: {path: 'nist'}
```

Comment:
currently all of them are ra-5, si-2 - if this is intentional, just hardcode that array in here.  if you can find nist tags for them, then your code is buggy